packages/std/src/math/uint128.rs

```diff
@@ -232,10 +233,14 @@ impl Uint128 {
         if denominator == 0 {
             panic!("Denominator must not be zero");
         }
-        // TODO: avoid overflow in multiplication (https://github.com/CosmWasm/cosmwasm/issues/920)
-        let val = self.u128() * numerator / denominator;
+        let val = (self.full_mul(numerator) / denominator).as_u128();
```

Comment:
It would be nice to be able to make the panicking explicit and visible here as well. Then we'd controll the message.

Is there a TryFrom implementation `U256` -> `u128`? I can't find it in the docs but there is code here: https://github.com/paritytech/parity-common/blob/61fc474c4f14d2291820282aed081256c2391510/uint/src/uint.rs#L536-L549

Reply:
Looks like there is one after all!